templates/AlgoliaSearch/src/app.js.hbs

```diff
@@ -0,0 +1,28 @@
+import algoliasearch from 'algoliasearch';
+
+const client = algoliasearch('{{appId}}', '{{apiKey}}');
+const index = client.initIndex('{{indexName}}');
+
+const $searchBox = document.querySelector('#searchBox input[type=search]');
+const $hits = document.querySelector('#hits');
+
+function renderHits(query) {
+  index.search(query).then(result => {
+    $hits.innerHTML = `
+      <ul class="ais-hits">
+        ${result.hits
+          .map(hit => `<li class="ais-hits--item">{{#if mainAttribute}}${hit._highlightResult.{{mainAttribute}}.value}{{else}}${JSON.stringify(hit)}{{/if}}</li>`)
```

Comment:
I'd add a comment here that this is vulnerable to XSS, and if people have user-provided data it should be sanitized

Reply:
Because of the `innerHTML`?

Reply:
yes, innerHTML leaves this vulnerable, just a comment would be enough warning IMO